File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/javadoc/uncomment/direct/Java/Python/s863488345_commented.py
import sys

def main():
    input = sys.stdin.read().split()
    index = 0
    mapNum = int(input[index])
    index += 1

    for _ in range(mapNum):
        if _ > 0:
            print("")
        h = int(input[index])
        w = int(input[index + 1])
        index += 2

        battleTown = []
        for _ in range(h):
            row = list(input[index])
            index += 1
            battleTown.append(row)

        x, y = -1, -1
        for i in range(h):
            for j in range(w):
                if battleTown[i][j] in ['^', 'v', '<', '>']:
                    x, y = i, j

        bomberActNum = int(input[index])
        bomberAct = list(input[index + 1])
        index += 2

        if 0 <= x < h and 0 <= y < w:
            for action in bomberAct:
                if action == 'U':
                    battleTown[x][y] = '^'
                    if x - 1 >= 0 and battleTown[x - 1][y] == '.':
                        battleTown[x - 1][y] = '^'
                        battleTown[x][y] = '.'
                        x -= 1
                elif action == 'D':
                    battleTown[x][y] = 'v'
                    if x + 1 < h and battleTown[x + 1][y] == '.':
                        battleTown[x + 1][y] = 'v'
                        battleTown[x][y] = '.'
                        x += 1
                elif action == 'L':
                    battleTown[x][y] = '<'
                    if y - 1 >= 0 and battleTown[x][y - 1] == '.':
                        battleTown[x][y - 1] = '<'
                        battleTown[x][y] = '.'
                        y -= 1
                elif action == 'R':
                    battleTown[x][y] = '>'
                    if y + 1 < w and battleTown[x][y + 1] == '.':
                        battleTown[x][y + 1] = '>'
                        battleTown[x][y] = '.'
                        y += 1
                elif action == 'S':
                    xOfBomber, yOfBomber = x, y
                    moveBomber(battleTown, xOfBomber, yOfBomber, h, w)

            for row in battleTown:
                print(''.join(row))

def moveBomber(battleTown, a, b, h, w):
    if 0 <= a < h and 0 <= b < w:
        if battleTown[a][b] == '^':
            for i in range(a - 1, -1, -1):
                if battleTown[i][b] == '*':
                    battleTown[i][b] = '.'
                    return
                elif battleTown[i][b] == '#':
                    return
        elif battleTown[a][b] == 'v':
            for i in range(a + 1, h):
                if battleTown[i][b] == '*':
                    battleTown[i][b] = '.'
                    return
                elif battleTown[i][b] == '#':
                    return
        elif battleTown[a][b] == '<':
            for j in range(b - 1, -1, -1):
                if battleTown[a][j] == '*':
                    battleTown[a][j] = '.'
                    return
                elif battleTown[a][j] == '#':
                    return
        elif battleTown[a][b] == '>':
            for j in range(b + 1, w):
                if battleTown[a][j] == '*':
                    battleTown[a][j] = '.'
                    return
                elif battleTown[a][j] == '#':
                    return

# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/javadoc/uncomment/direct/Java/Go/s863488345_commented.go
package main

import (
	"fmt"
)

var (
	mapNum     int
	h, w       int
	battleTown [][]rune
	bomberActNum int
	bomberAct    []rune
	tank         [2]int
	xOfTank, yOfTank int
	xOfBomber, yOfBomber int
)

func main() {
	var sc *Scanner = NewScanner()
	mapNum = sc.nextInt()

	for i := 0; i < mapNum; i++ {
		if i > 0 {
			fmt.Println()
		}
		h = sc.nextInt()
		w = sc.nextInt()

		battleTown = make([][]rune, h)
		for j := range battleTown {
			battleTown[j] = make([]rune, w)
			line := sc.next()
			for k, char := range line {
				battleTown[j][k] = char

				if char == '^' || char == 'v' || char == '<' || char == '>' {
					xOfTank = j
					yOfTank = k
				}
			}
		}

		bomberActNum = sc.nextInt()
		bomberAct = []rune(sc.next())

		if 0 <= xOfTank && xOfTank < h && 0 <= yOfTank && yOfTank < w {
			for l := 0; l < bomberActNum; l++ {
				switch bomberAct[l] {
				case 'U':
					battleTown[xOfTank][yOfTank] = '^'
					if xOfTank-1 >= 0 && battleTown[xOfTank-1][yOfTank] == '.' {
						battleTown[xOfTank-1][yOfTank] = '^'
						battleTown[xOfTank][yOfTank] = '.'
						xOfTank--
					}
				case 'D':
					battleTown[xOfTank][yOfTank] = 'v'
					if xOfTank+1 < h && battleTown[xOfTank+1][yOfTank] == '.' {
						battleTown[xOfTank+1][yOfTank] = 'v'
						battleTown[xOfTank][yOfTank] = '.'
						xOfTank++
					}
				case 'L':
					battleTown[xOfTank][yOfTank] = '<'
					if yOfTank-1 >= 0 && battleTown[xOfTank][yOfTank-1] == '.' {
						battleTown[xOfTank][yOfTank-1] = '<'
						battleTown[xOfTank][yOfTank] = '.'
						yOfTank--
					}
				case 'R':
					battleTown[xOfTank][yOfTank] = '>'
					if yOfTank+1 < w && battleTown[xOfTank][yOfTank+1] == '.' {
						battleTown[xOfTank][yOfTank+1] = '>'
						battleTown[xOfTank][yOfTank] = '.'
						yOfTank++
					}
				case 'S':
					xOfBomber = xOfTank
					yOfBomber = yOfTank
					moveBomber(xOfBomber, yOfBomber)
				}
			}
			for s := 0; s < h; s++ {
				for k := 0; k < w; k++ {
					fmt.Printf("%c", battleTown[s][k])
				}
				fmt.Println()
			}
		}
	}
}

func moveBomber(a, b int) {
	if 0 <= a && a < h && 0 <= b && b < w {
		switch battleTown[a][b] {
		case '^':
			for i := a - 1; i >= 0; i-- {
				if battleTown[i][b] == '*' {
					battleTown[i][b] = '.'
					return
				} else if battleTown[i][b] == '#' {
					return
				}
			}
		case 'v':
			for i := a + 1; i < h; i++ {
				if battleTown[i][b] == '*' {
					battleTown[i][b] = '.'
					return
				} else if battleTown[i][b] == '#' {
					return
				}
			}
		case '<':
			for i := b - 1; i >= 0; i-- {
				if battleTown[a][i] == '*' {
					battleTown[a][i] = '.'
					return
				} else if battleTown[a][i] == '#' {
					return
				}
			}
		case '>':
			for i := b + 1; i < w; i++ {
				if battleTown[a][i] == '*' {
					battleTown[a][i] = '.'
					return
				} else if battleTown[a][i] == '#' {
					return
				}
			}
		}
	}
}

type Scanner struct {
	r   []rune
	i   int
	n   int
	err error
}

func NewScanner() *Scanner {
	s := &Scanner{}
	s.r = []rune{}
	s.i = 0
	s.n = 0
	s.err = nil
	return s
}

func (s *Scanner) next() string {
	if s.err != nil {
		return ""
	}
	for s.i >= s.n {
		var buf []byte
		buf, s.err = fmt.Scanln()
		if s.err != nil {
			return ""
		}
		s.r = []rune(string(buf))
		s.i = 0
		s.n = len(s.r)
	}
	start := s.i
	for s.i < s.n && s.r[s.i] != ' ' {
		s.i++
	}
	return string(s.r[start:s.i])
}

func (s *Scanner) nextInt() int {
	str := s.next()
	num := 0
	for _, c := range str {
		num = num*10 + int(c-'0')
	}
	return num
}

// End of Code
